Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2113,7 +2113,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             let copyAction = UIAlertAction(title: copyImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction) -> Void in
                 let pasteBoard = UIPasteboard.generalPasteboard()
                 pasteBoard.string = url.absoluteString
-                // TODO: put the actual image on the clipboard
+                // put the actual image on the clipboard
+                // do this be asyncronously just in case we're in a low bandwidth situation
+                // Is this neccessary, or is the image already sat in cache?
+                NSURLSession.sharedSession().dataTaskWithURL(url, completionHandler: { (data, _, error) -> Void in
```

Comment:
@sleroux: I've now added background handler, and testing it on Very Bad Network using the Link Conditioner, we end up in a situation whereby we have a URL copied into the clipboard until the image has downloaded. But we provide no indication to let the user know that the image download has been successful, and so it is pot luck whether what you paste is the image or the link.

Safari seems to grab it's images from cache as even with link conditioner on, there is no time lag between pressing copy on an image and the image being added to the pasteboard. 
Safari has access to the offline cache, which is a private API and therefore something we can't use ourselves.

Chrome does not provide a Copy Image option.

The only way I can know how to achieve that is to use JavaScript to draw the image to a Canvas and then somehow getting that canvas image back out of the JS and into swift to add to the clipboard. Is that even a sensible thing to do?

Alternatively, we can maybe send a notification to the user to let them know that the image has been copied to clipboard, or do the copying synchronously with some kind of progress bar. I can't quite tell which solution would be more annoying for the user, they are all pretty crap.

Any other ideas?